airflow/utils/callback_requests.py

```diff
@@ -41,8 +54,12 @@ def __eq__(self, other):
     def __repr__(self):
         return str(self.__dict__)
 
+    def execute(self, dagbag: "DagBag", *, session: Session) -> None:
+        """Execute the callback."""
+        raise NotImplementedError("implement in subclass")
+
 
-class TaskCallbackRequest(CallbackRequest):
+class TaskCallbackRequest(CallbackRequest, LoggingMixin):
```

Comment:
This is _probably_ safe, but these objects are passed over a multiprocessing Pipe, so are subject to pickling, so we should create a `__getstate__`(?) method to ensure that `self.log` is never attempted to be pickled.

Reply:
Alternatively I can pass in the logger instance as an argument of `execute()` instead? I don’t really like making CallbackRequest instances subclass from `LoggingMixin` tbh, but passing a logger around also feels a bit weird.